core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt

```diff
@@ -73,12 +78,31 @@ class CommonDiscriminator(private val discriminator: String) : Discriminator {
 
     private fun removeDiscriminatorEntry(jsonObject: JsonObject): JsonObject {
         return jsonObject.entries
-            .toMutableSet()
+            .toMutableList()
```

Comment:
Did not get the reason why set did not work here? Can you explain in more detail?

Reply:
I modified the function to handle the extraction and reinsertion of meta and links without changing their positions. The use of `Set` was avoided to maintain the original order and prevent a non-standard JSON format, which would have otherwise pushed meta and links to the end of the document and required updates to our test JSON files.

https://github.com/infinum/kotlin-jsonapix/blob/94de42b88b13e21fee5d79e996ccdcc6d14fd84f/core/src/main/java/com/infinum/jsonapix/core/discriminators/CommonDiscriminator.kt#L96-L105